pkg/config/protocol/instance.go

```diff
@@ -56,6 +56,12 @@ const (
 	Unsupported Instance = "UnsupportedProtocol"
 )
 
+// ProtocolOrder defines precedence of protocol in case there is protocol conflict for a port.
+var ProtocolOrder = map[Instance]int{
+	UDP: 1, Unsupported: 2, TCP: 3, TLS: 4, HTTPS: 5, MySQL: 6, Redis: 7, Mongo: 8,
+	Thrift: 9, HTTP: 10, HTTP_PROXY: 11, HTTP2: 12, GRPC: 13, GRPCWeb: 14,
```

Comment:
It looks confusing about this, should we downgrade all ports to tcp if they are not equal

Reply:
If user have http2 and grpc, which works perfectly fine before, and should not be downgraded.

I agree this is confusing, and I cannot find a clear way to set protocol precedences. Probably we should go back to the original commit to just remove protocol from target port dedup: https://github.com/istio/istio/pull/36462/commits/7e9835f29aa9889d0fbfd1baa36c4a2d24cc9676. It's a clean change and makes user service behave consistently, which is strictly better than before where service could be broken or work in different ways randomly. @howardjohn wdyt?

Reply:
SG, as this is only working on inbound, which does not have much diff

Reply:
Reverted to [7e9835f](https://github.com/istio/istio/commit/7e9835f29aa9889d0fbfd1baa36c4a2d24cc9676).